src/agent/agent.js

```diff
@@ -77,18 +80,46 @@ export class Agent {
                 this.handleMessage('system', init_message, 2);
             }
             else {
-                this.bot.chat('Hello world! I am ' + this.name);
+		const translation = await this.handleTranslation("Hello world! I am", `${preferred_lang}`);
+		print(translation)
+                this.bot.chat(translation+" "+this.name);
                 this.bot.emit('finished_executing');
+
             }
 
             this.startEvents();
         });
     }
 
-    cleanChat(message) {
+    async handleTranslation(message, lang) {
```

Comment:
handleTranslation should be in its own file, `translator.js` maybe, and be imported where needed.
You shouldn't have to pass prefered_language to it, it should know by reading settings what language to translate to. maybe have a separate function to translate specifically to english